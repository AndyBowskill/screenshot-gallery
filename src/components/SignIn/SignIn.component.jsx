import React from 'react';

import './SignIn.style.css';

class SignIn extends React.Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
    };
  }

  onEmailChange = (event) => {
    this.setState({ email: event.target.value });
  };

  onPasswordChange = (event) => {
    this.setState({ password: event.target.value });
  };

  onSignInSumbit = () => {
    fetch('http://localhost:3000/signin', {
      method: 'post',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password,
      }),
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.user) {
          this.props.loadUser(data.user);
          this.props.loadScreenshots(data.screenshots);
          this.props.onRouteChange('home');
        }
      });
  };

  render() {
    return (
      <section className='signin flex'>
        <div>
          <h3>Sign In</h3>

          <label htmlFor='email'>Email</label>
          <input
            type='text'
            placeholder='Please enter email'
            name='email'
            id='email'
            required
            onChange={this.onEmailChange}
          />

          <label htmlFor='psw'>Password</label>
          <input
            type='password'
            placeholder='Please enter password'
            name='psw'
            id='psw'
            required
            onChange={this.onPasswordChange}
          />

          <button type='submit' onClick={this.onSignInSumbit}>
            Sign In
          </button>
        </div>
      </section>
    );
  }
}

export default SignIn;
