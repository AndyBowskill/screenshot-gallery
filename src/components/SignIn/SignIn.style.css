.signin.flex {
  display: flex;
  justify-content: center;
  padding: 16px;
}

.signin.flex h3 {
  font-size: 1rem;
}

.signin.flex input {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  border: 1px solid #0074d9;
  border-radius: 5px;
  width: 340px;
  display: block;
  box-sizing: border-box;
  padding: 16px;
  margin: 12px 0 12px 0;
}

.signin.flex button {
  font-family: 'Didact Gothic', sans-serif;
  font-size: 1rem;
  background-color: rgb(127, 219, 255);
  color: #0074d9;
  padding: 12px 30px;
  margin: 12px 0 24px 0;
  text-align: center;
  text-decoration: none;
  border: 1px solid #0074d9;
  cursor: pointer;
  border-radius: 5px;
  width: 340px;
  display: block;
  opacity: 0.9;
}

.signin.flex hr {
  height: 1px;
  background-color: #0074d9;
  border: none;
  margin-bottom: 22px;
}

@media only screen and (min-width: 480px) {
  .signin.flex h3 {
    font-size: 1.2rem;
  }
}
