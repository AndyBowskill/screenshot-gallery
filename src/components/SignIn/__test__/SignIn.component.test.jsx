import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../SignIn.component';

const onRouteChange = jest.fn();
const loadUser = jest.fn();
const loadScreenshots = jest.fn();

test('SignIn component displays correctly.', () => {
  const component = renderer.create(
    <SignIn
      onRouteChange={onRouteChange}
      loadUser={loadUser}
      loadScreenshots={loadScreenshots}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
