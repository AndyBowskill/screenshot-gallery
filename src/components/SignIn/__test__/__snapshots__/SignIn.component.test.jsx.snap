// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component displays correctly. 1`] = `
<section
  className="signin flex"
>
  <div>
    <h1>
      Sign In
    </h1>
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Please enter email"
      required={true}
      type="text"
    />
    <label
      htmlFor="psw"
    >
      Password
    </label>
    <input
      id="psw"
      name="psw"
      onChange={[Function]}
      placeholder="Please enter password"
      required={true}
      type="password"
    />
    <button
      onClick={[Function]}
      type="submit"
    >
      Sign In
    </button>
  </div>
</section>
`;
